### PR TITLE
Pango odds and ends

### PR DESCRIPTION
**Pull request by jf2048**
The only thing missing now is the C bitfields

### CODE REVIEW

File: glib/src/auto/markup_parse_context.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    match fn {
+        ref => |ptr| ffi::g_markup_parse_context_ref(ptr),
+        unref => |ptr| ffi::g_markup_parse_context_unref(ptr),
+        type_ => || ffi::g_markup_parse_context_get_type(),
+    }
+}
+
+impl MarkupParseContext {
+    //#[doc(alias = "g_markup_parse_context_new")]
+    //pub fn new(parser: /*Ignored*/&MarkupParser, flags: /*Ignored*/MarkupParseFlags, user_data: /*Unimplemented*/Option<Basic: Pointer>) -> MarkupParseContext {
```
Comment (sdroege) [inline glib/src/auto/markup_parse_context.rs:23]: Is it useful without the constructor?

File: pango/src/glyph_string.rs [hunk 2]
```diff
@@ -32,4 +32,63 @@ impl GlyphString {
             }
         }
     }
+
+    pub fn log_clusters(&self) -> &[i32] {
+        unsafe {
+            let ptr = (*self.to_glib_none().0).log_clusters as *const i32;
```
Comment (sdroege) [inline pango/src/glyph_string.rs:38]: https://gtk-rs.org/gtk-rs-core/git/docs/glib/collections/struct.Slice.html#method.from_glib_borrow_num if you feel like it :)

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: pango/src/script_iter.rs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+impl<'text> glib::StaticType for ScriptIter<'text> {
+    #[inline]
+    fn static_type() -> glib::Type {
+        unsafe { from_glib(ffi::pango_script_iter_get_type()) }
+    }
+}
+
+impl<'text> ScriptIter<'text> {
+    #[doc(alias = "pango_script_iter_new")]
+    pub fn new(text: impl AsRef<GStr> + 'text) -> Self {
```
Comment (sdroege) [inline pango/src/script_iter.rs:53]: Do we want to change this here to `AsRef<GStr>` already, or defer that to next release with the other related changes?

Note: only for parameters this is potentially problematic

File: glib/src/auto/markup_parse_context.rs (hunk 1 above)
Comment (jf2048) [reply glib/src/auto/markup_parse_context.rs:23]: Yes, you can create one with `pango_markup_parser_new` which is why I included it here. I can try to bind `g_markup_parse_context_new` in another PR

File: pango/src/glyph_string.rs (hunk 2 above)
Comment (jf2048) [reply pango/src/glyph_string.rs:31]: Fixed, and added `Slice::from_glib_borrow_num_mut` too

File: pango/src/script_iter.rs (hunk 3 above)
Comment (jf2048) [reply pango/src/script_iter.rs:53]: I think this one should stay `AsRef<GStr>` for now, with an owned `GString` it forces the `Iterator` implementation to copy the string every time.

With `IntoGStr` we can fix this and also optimize out those `GStr::from_ptr` calls.

Comment (sdroege) [reply pango/src/script_iter.rs:53]: Works for me

**Review by sdroege — DISMISSED** [review]

**Review by sdroege — APPROVED** [review]

**Pull request merged**